aquarius/events/events_monitor.py

```diff
@@ -119,60 +118,83 @@ def __init__(self, web3, config_file):
             "Purgatory enabled" if self.purgatory else "Purgatory disabled"
         )
         logger.info(purgatory_message)
+        self._thread_process_blocks_is_on = False
+        self._thread_process_queue_is_on = False
+        self._thread_process_ve_allocate_is_on = False
+        self._thread_process_purgatory_is_on = False
 
     @property
     def block_envvar(self):
         return "METADATA_CONTRACT_BLOCK"
 
-    def start_events_monitor(self):
-        if self._monitor_is_on:
-            return
+    def stop_monitor(self):
+        """Stops all threads for processing more data"""
+        self._thread_process_blocks_is_on = False
+        self._thread_process_queue_is_on = False
+        self._thread_process_ve_allocate_is_on = False
+        self._thread_process_purgatory_is_on = False
```

Comment:
may be I am missing something but shouldn't it include self._monitor_is_on = False?

Reply:
no, I have removed _monitor_is_on and replaced it with 4 variables, one for each thread